manticore/native/state.py

```diff
@@ -25,27 +58,34 @@ def execute(self):
             ConcretizeRegister,
         )  # must be here, otherwise we get circular imports
 
+        self._checkpoint()
         try:
             result = self._platform.execute()
 
         # Instead of State importing SymbolicRegisterException and SymbolicMemoryException
         # from cpu/memory shouldn't we import Concretize from linux, cpu, memory ??
         # We are forcing State to have abstractcpu
-        except ConcretizeRegister as e:
+        except ConcretizeRegister as exc:
+            # Need to define local variable to use in closure
+            e = exc
             expression = self.cpu.read_register(e.reg_name)
 
```

Comment:
As every child state will do the rollback, can not we roll it back before raising the Concretize?

Reply:
I think https://github.com/trailofbits/manticore/pull/1697/commits/f933e585cdf93af948db3aa98ba5b3ec2b6efd04 should fix this. Thank you for bringing that up